js/stdlib/ts/ALGO.ts

```diff
@@ -2141,52 +2141,66 @@ export function unsafeGetMnemonic(acc: NetworkAccount|Account): string {
 }
 
-export async function launchToken (accCreator:Account, name:string, sym:string, opts:LaunchTokenOpts = {}) {
-  debug(`Launching token, ${name} (${sym})`);
-  const addr = (acc:Account) => acc.networkAccount.addr;
-  const caddr = addr(accCreator);
-  const zaddr = caddr;
-  // ^ XXX should be nothing; docs say can be "", but doesn't actually work
-  const client = await getAlgodClient();
-  const dotxn = async (mktxn:(params:TxnParams) => Transaction, acc:Account = accCreator) => {
-    const sk = acc.networkAccount.sk;
-    if ( ! sk ) {
-      throw new Error(`can only launchToken with account with secret key`);
-    }
-    const params = await getTxnParams('launchToken');
-    const t = mktxn(params);
-    const s = t.signTxn(sk);
-    const r = (await client.sendRawTransaction(s).do());
-    return await waitForConfirmation(r.txId);
-  };
+const makeAssetCreateTxn = (
+  creator: Address,
+  supply: BigNumber,
+  decimals: number,
+  symbol: string,
+  name: string,
+  url: string,
+  metadataHash: string,
+  clawback: Address,
+  params: TxnParams,
+): Transaction => {
+  return algosdk.makeAssetCreateTxnWithSuggestedParamsFromObject({
+    from: creator,
+    total: bigNumberToBigInt(supply),
+    decimals: decimals,
+    defaultFrozen: false,
+    unitName: symbol,
+    assetName: name,
+    assetURL: url,
+    assetMetadataHash: metadataHash,
+    clawback: clawback,
+    suggestedParams: params,
+  });
+}
+
+export const launchToken = async (accCreator: Account, name: string, sym: string, opts: LaunchTokenOpts = {}) => {
+  const addrCreator = accCreator.networkAccount.addr;
   const supply = opts.supply ? bigNumberify(opts.supply) : bigNumberify(2).pow(64).sub(1);
-  const decimals = opts.decimals !== undefined ? opts.decimals : 6;
-  const clawback = opts.clawback !== undefined ? addr(opts.clawback) : zaddr;
-  const ctxn_p = await dotxn(
-    (params:TxnParams) =>
-    algosdk.makeAssetCreateTxnWithSuggestedParams(
-      caddr, undefined, bigNumberToBigInt(supply), decimals,
-      false, zaddr, zaddr, zaddr, clawback,
-      sym, name, '', '', params,
-    ));
-  const idn = ctxn_p['created-asset-index'];
-  if ( ! idn ) { throw Error(`${sym} no asset-index!`); }
-  const id = bigNumberify(idn);
-  debug(`${sym}: asset is ${id}`);
-
-  const mint = async (accTo:Account, amt:unknown) => {
-    debug(`${sym}: transferring ${amt} ${sym} for ${addr(accTo)}`);
-    await transfer(accCreator, accTo, amt, id);
-  };
-  const optOut = async (accFrom:Account, accTo:Account = accCreator) => {
-    await dotxn(
-      (params) =>
-      algosdk.makeAssetTransferTxnWithSuggestedParams(
-        addr(accFrom), addr(accTo), addr(accTo), undefined,
-        0, undefined, idn, params
-      ), accFrom);
-  };
+  const decimals = opts.decimals ?? 6;
+  const url = opts.url ?? '';
+  const metadataHash = opts.metadataHash ?? '';
+  const clawback = opts.clawback ? opts.clawback.networkAccount.addr : '';
```

Comment:
No, the clawback should be an address. You should use `protect(T_Address, opts.clawback)` to get a canonical form and that will be able to do with addresses and with network accounts